tests/test_callbacks.py

```diff
@@ -0,0 +1,153 @@
+        create_prompt,
+        remove_columns=[
+            "question",
+            "context",
+            "long_answer",
+            "pubid",
+            "final_decision",
+        ],
+    )
+
+    def tokenize_text(dataset, padding="max_length"):
+        model_input = tokenizer(dataset["input"], **tokenizer_args)
+        response = tokenizer(dataset["response"], **tokenizer_args)
+
+        if padding == "max_length":
+            response["input_ids"] = [
+                [(l if l != tokenizer.pad_token_id else -100) for l in label]
+                for label in response["input_ids"]
+            ]
+
+        model_input["labels"] = response["input_ids"]
+        return model_input
+
+    tokenized_dataset = merged_dataset.map(
+        tokenize_text, remove_columns=["input", "response"]
+    )
+    tokenized_dataset = tokenized_dataset.map(
+        lambda x: {
+            "input_ids": x["input_ids"][0],
+            "labels": x["labels"][0],
+            "attention_mask": x["attention_mask"][0],
+        }
+    )
+    return dataset, merged_dataset, tokenized_dataset
+
+
+def create_deepeval_dataset(dataset, sample_size):
+    eval_dataset = [dataset[row] for row in range(5, 10)]
+    goldens = []
+    for row in eval_dataset:
+        golden = Golden(
+            input=row["question"],
+            expectedOutput=row["long_answer"],
+            context=row["context"]["contexts"],
+            retrieval_context=row["context"]["contexts"],
+        )
+        goldens.append(golden)
+
+    return EvaluationDataset(goldens=goldens)
+
+
+if __name__ == "__main__":
+    # initialize tokenizer
+    tokenizer = T5Tokenizer.from_pretrained("google/flan-t5-small")
+
+    # initalize model
+    model = T5ForConditionalGeneration.from_pretrained("google/flan-t5-small")
+    model.resize_token_embeddings(len(tokenizer))
+
+    # create tokenized dataset
+    tokenizer_args = {
+        "return_tensors": "pt",
+        "max_length": 128,
+        "padding": "max_length",
+        "truncation": True,
+        "padding": True,
+    }
+
+    dataset, merged_dataset, tokenized_dataset = prepare_dataset(
+        tokenizer, tokenizer_args
+    )
+
+    label_pad_token_id = -100
+    data_collator = DataCollatorForSeq2Seq(
+        tokenizer,
+        model=model,
+        label_pad_token_id=label_pad_token_id,
+        pad_to_multiple_of=8,
+    )
+
+    repository_id = f"flan-t5-small"
+
+    # Define training args
+    training_args = Seq2SeqTrainingArguments(
+        output_dir=repository_id,
+        overwrite_output_dir=True,
+        num_train_epochs=50,
+        per_device_train_batch_size=8,
+    )
+
+    # Create Trainer instance
+    trainer = Seq2SeqTrainer(
+        model=model,
+        tokenizer=tokenizer,
+        args=training_args,
+        data_collator=data_collator,
+        train_dataset=tokenized_dataset["train"],
+    )
+
+    eval_dataset = create_deepeval_dataset(dataset["train"], sample_size=5)
```

Comment:
Is there a type definition for dataset["train"] for me to dig into? Or is it different depending on the dataset

Reply:
Yeah it depends on the dataset, I am using 🤗's `datasets` library for loading and stuff (https://github.com/huggingface/datasets) which uses this type of partitioning.